presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -588,120 +590,111 @@ public PlanWithProperties visitTopNRowNumber(TopNRowNumberNode node, StreamPrefe
         //
 
         @Override
-        public PlanWithProperties visitTableWriter(TableWriterNode originalTableWriterNode, StreamPreferredProperties parentPreferences)
+        public PlanWithProperties visitTableWriter(TableWriterNode tableWrite, StreamPreferredProperties parentPreferences)
```

Comment:
maybe add some comments before each block explaining what that case is for, as this method is a bit complex.